oslabs-beta/CodeMapper
# Initial file-walking, Babel Parsing & Traversing, and README. Bonus: Sample AST!

**Getting some initial things in place with this pull request.**

Here's what I changed:
1. Added the initial package.json and package-lock.json files.
2. Added an initial README to describe what's I'm adding to the project.
3. Added our initial filewalker that lets us iterate through a file and folder structure.
4. Added our initial babel parser which generates an AST from a JS file, and the traversing "visitor" function which can iterate over the AST and interact with it based on what we define in the visitor.
5. Added a sample AST for convenience, which is generated based on the filewalker.js file also in this repo.

How to test:
1. Be sure to run npm install when testing this PR!

2. To try out the filewalker, use the "generateResult" function at the very end, being sure to adjust the options if you'd like to get different options in the resulting list or tree, and putting in your own folder path to start with as the argument to "generateResult" at the very end of the file.

It's best tested with a nested file/folder structure that has at least a couple files and folders in it.
We also currently ignore node_modules by default, but all inclusion or exclusion settings will come from the user in future rather than being hard-coded into the file.

3. To try out the babel parser and traverser, please note that the visitor doesn't do much at the moment - that's the piece we're currently testing out. We also haven't necessarily decided 100% on the Babel parser specifically, but it seems like a good option.

In line 12, it generates an AST based on the file you give it in lines 6 & 10.

After that the traverser is invoked with the traverse function, passing in the parsedFile and the visitor function so it can traverse the tree. 

The AST for the file will be written to a data.json file in the same folder as the babelParse file.
